static/app/views/automations/components/ruleNode.tsx

```diff
@@ -0,0 +1,88 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import {Flex} from 'sentry/components/container/flex';
+import {Button} from 'sentry/components/core/button';
+import {IconDelete} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {DataCondition} from 'sentry/types/workflowEngine/dataConditions';
+import {
+  DataConditionNodeContext,
+  dataConditionNodesMap,
+  useDataConditionNodeContext,
+} from 'sentry/views/automations/components/dataConditionNodes';
+
+interface RuleNodeProps {
+  condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>;
+  condition_id: string;
+  onDelete: () => void;
+  onUpdate: (comparison: Record<string, any>) => void;
+}
+
+function Node() {
+  const {condition} = useDataConditionNodeContext();
+  const node = dataConditionNodesMap[condition.comparison_type];
+
+  if (node?.configNode) {
+    return node.configNode();
+  }
+
+  return <Fragment>{node?.label}</Fragment>;
+}
+
+export default function RuleNode({
+  condition,
+  condition_id,
+  onDelete,
+  onUpdate,
+}: RuleNodeProps) {
+  return (
+    <RuleRowContainer>
+      <RuleRow>
+        <Rule>
+          <DataConditionNodeContext.Provider value={{condition, condition_id, onUpdate}}>
+            <Node />
+          </DataConditionNodeContext.Provider>
+        </Rule>
+        <DeleteButton
+          aria-label={t('Delete Node')}
+          size="sm"
+          icon={<IconDelete />}
+          borderless
+          onClick={onDelete}
+        />
+      </RuleRow>
+    </RuleRowContainer>
+  );
+}
+
+const RuleRowContainer = styled('div')<{incompatible?: boolean}>`
+  background-color: ${p => p.theme.backgroundSecondary};
+  border-radius: ${p => p.theme.borderRadius};
+  border: 1px ${p => p.theme.innerBorder} solid;
+  border-color: ${p => (p.incompatible ? p.theme.red200 : 'none')};
+`;
+
+const RuleRow = styled(Flex)`
+  align-items: center;
```

Comment:
avoid styling flex. mixing where the styles are declared and creates two styled elements